src/LibraryManager/Manifest.cs

```diff
@@ -196,17 +211,16 @@ public Manifest Clone()
         }
 
         /// <summary>
-        /// Installs a library with the given libraryId
+        /// Validates a single library 
         /// </summary>
         /// <param name="libraryId"></param>
         /// <param name="providerId"></param>
         /// <param name="files"></param>
         /// <param name="destination"></param>
         /// <param name="cancellationToken"></param>
         /// <returns></returns>
-        public async Task<IEnumerable<ILibraryOperationResult>> InstallLibraryAsync(string libraryId, string providerId, IReadOnlyList<string> files, string destination, CancellationToken cancellationToken)
+        public async Task<ILibraryOperationResult> ValidateLibraryAsync(string libraryId, string providerId, IReadOnlyList<string> files, string destination, CancellationToken cancellationToken)
```

Comment:
I was looking at work item: #148 and what can be consumed from here for the UI work.

So do we call this every time there is an invalid entry in the UI? Say user enters library in the correct format for cdnjs-@ and then deletes version, do we intend to call this method for validation?

We need to do all validation before enabling Install button. I am having trouble understanding what can be consumed from these changes. Could you please help me here?

Thanks!

Reply:
I sent you an IM with some ideas to do without this PR.